File: rebrand.py
import os
import re

# Define mappings
replacements = {
    r'\bemerald-500\b': 'msbm-red',
    r'\bemerald-600\b': 'msbm-red',
    r'\bemerald-700\b': 'msbm-red',
    r'\bteal-500\b': 'inner-blue',
    r'\bteal-600\b': 'inner-blue',
    r'\bemerald-100\b': 'msbm-red/10',
    r'\bteal-100\b': 'inner-blue/10',
    r'\bemerald-50\b': 'msbm-red/5',
    r'\bteal-50\b': 'inner-blue/5',
    r'\bemerald-400\b': 'msbm-red-bright',
    r'\bemerald-300\b': 'msbm-red-bright',
    r'\bteal-400\b': 'light-blue',
    r'\bteal-300\b': 'light-blue',
    r'\bemerald-900/30\b': 'msbm-red/20',
    r'\bteal-900/30\b': 'inner-blue/20',
    r'\bemerald-950/20\b': 'msbm-red/10',
    r'\bemerald-950/30\b': 'msbm-red/15',
    r'\bemerald-950/10\b': 'msbm-red/5',
    r'\bshadow-emerald-500/25\b': 'shadow-msbm-red/25',
    r'\bfrom-emerald-500\b': 'from-msbm-red',
    r'\bto-teal-500\b': 'to-inner-blue',
    r'\bfrom-emerald-600\b': 'from-msbm-red',
    r'\bto-teal-600\b': 'to-inner-blue',
    r'\bvia-emerald-950\b': 'via-[#4a0a10]',
    r'\bvia-teal-600\b': 'via-inner-blue',
    r'\bto-emerald-700\b': 'to-msbm-red',
    r'\bto-emerald-50/30\b': 'to-msbm-red/5',
}

def replace_in_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    new_content = content
    for pattern, replacement in replacements.items():
        new_content = re.sub(pattern, replacement, new_content)
    
    if new_content != content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated: {file_path}")

def walk_and_replace(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.tsx', '.ts', '.css')):
                replace_in_file(os.path.join(root, file))

if __name__ == "__main__":
    src_dir = r"c:\Users\JT\Documents\WebDev Refresh\MSBM\MSBM-HRM-Suite\src"
    walk_and_replace(src_dir)

File: rebrand.js
/* eslint-disable @typescript-eslint/no-require-imports */
const fs = require('fs');
const path = require('path');

const directoryPath = path.join(__dirname, 'src', 'components', 'hrm');

const replaceColors = (content) => {
  // Regex patterns to match Tailwind classes and replace them
  // emerald-500 -> msbm-red
  // teal-600 -> inner-blue
  // emerald-50 -> msbm-red/5
  // emerald-900 -> msbm-red/20
  // emerald-950 -> msbm-red/30
  // emerald-100 -> msbm-red/10
  // emerald-200 -> msbm-red/20
  // emerald-300 -> msbm-red-bright
  // emerald-400 -> msbm-red
  // emerald-600 -> msbm-red
  // emerald-700 -> msbm-red
  // emerald-800 -> msbm-red
  
  let newContent = content;

  // General emerald/teal to msbm-red/inner-blue replacements
  newContent = newContent.replace(/from-emerald-\d+/g, 'from-msbm-red');
  newContent = newContent.replace(/to-teal-\d+/g, 'to-inner-blue');
  newContent = newContent.replace(/hover:from-emerald-\d+/g, 'hover:from-msbm-red-bright');
  newContent = newContent.replace(/hover:to-teal-\d+/g, 'hover:to-light-blue');

  // Text colors
  newContent = newContent.replace(/text-emerald-\d+/g, 'text-msbm-red');
  newContent = newContent.replace(/text-teal-\d+/g, 'text-inner-blue');

  // Border colors
  newContent = newContent.replace(/border-emerald-\d+\/\d+/g, 'border-msbm-red/20');
  newContent = newContent.replace(/border-emerald-\d+/g, 'border-msbm-red/20');
  newContent = newContent.replace(/border-teal-\d+/g, 'border-inner-blue/20');

  // BG colors
  newContent = newContent.replace(/bg-emerald-50\/(\d+)/g, 'bg-msbm-red/5');
  newContent = newContent.replace(/bg-emerald-50/g, 'bg-msbm-red/5');
  newContent = newContent.replace(/bg-emerald-100/g, 'bg-msbm-red/10');
  newContent = newContent.replace(/bg-emerald-900\/50/g, 'bg-msbm-red/20');
  newContent = newContent.replace(/bg-emerald-900\/40/g, 'bg-msbm-red/20');
  newContent = newContent.replace(/bg-emerald-950\/30/g, 'bg-msbm-red/30');
  newContent = newContent.replace(/bg-emerald-950\/20/g, 'bg-msbm-red/20');
  newContent = newContent.replace(/bg-emerald-950\/40/g, 'bg-msbm-red/30');
  newContent = newContent.replace(/bg-emerald-\d+/g, 'bg-msbm-red');
  newContent = newContent.replace(/bg-teal-\d+/g, 'bg-inner-blue');

  // Hover BG colors
  newContent = newContent.replace(/hover:bg-emerald-\d+\/\d+/g, 'hover:bg-msbm-red/10');
  newContent = newContent.replace(/hover:bg-emerald-\d+/g, 'hover:bg-msbm-red/10');

  // Rings and Shadows
  newContent = newContent.replace(/ring-emerald-\d+/g, 'ring-msbm-red');
  newContent = newContent.replace(/shadow-emerald-\d+\/\d+/g, 'shadow-msbm-red/20');
  newContent = newContent.replace(/shadow-emerald-\d+/g, 'shadow-msbm-red');

  // Raw color words (like color="emerald")
  newContent = newContent.replace(/color="emerald"/g, 'color="msbm-red"');
  newContent = newContent.replace(/color="teal"/g, 'color="inner-blue"');

  // Timeline dots
  newContent = newContent.replace(/timeline-dot-emerald/g, 'timeline-dot-msbm-red');

  return newContent;
};

const processDirectory = (dirPath) => {
  const files = fs.readdirSync(dirPath);

  files.forEach(file => {
    const fullPath = path.join(dirPath, file);
    if (fs.statSync(fullPath).isDirectory()) {
      processDirectory(fullPath);
    } else if (fullPath.endsWith('.tsx') || fullPath.endsWith('.ts')) {
      const content = fs.readFileSync(fullPath, 'utf8');
      const newContent = replaceColors(content);
      if (content !== newContent) {
        fs.writeFileSync(fullPath, newContent, 'utf8');
        console.log(`Updated: ${fullPath}`);
      }
    }
  });
};

processDirectory(directoryPath);
console.log('Rebranding script completed.');
